From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:14:45 +0530
Subject: [PATCH 063/109] Cygwin: Add AArch64 support in config.guess,
 dfp.m4, and configure.ac

(cherry picked from commit 3f68a4b4c621c74a6c6ecb325a9997000cc4c157)
---
 config.guess  | 3 +++
 config/dfp.m4 | 4 ++--
 configure.ac  | 4 ++--
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/config.guess b/config.guess
index 1972fda8e..f7c9844b8 100755
--- a/config.guess
+++ b/config.guess
@@ -911,6 +911,9 @@ EOF
     i*:UWIN*:*)
 	echo "$UNAME_MACHINE"-pc-uwin
 	exit ;;
+    aarch64:CYGWIN*:*:*)
+	echo aarch64-pc-cygwin
+	exit ;;
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
diff --git a/config/dfp.m4 b/config/dfp.m4
index 5b29089ce..714bee6b2 100644
--- a/config/dfp.m4
+++ b/config/dfp.m4
@@ -22,8 +22,8 @@ Valid choices are 'yes', 'bid', 'dpd', and 'no'.]) ;;
   case $1 in
     powerpc*-*-linux* | i?86*-*-linux* | x86_64*-*-linux* | s390*-*-linux* | \
     i?86*-*-elfiamcu | i?86*-*-gnu* | \
-    i?86*-*-mingw* | x86_64*-*-mingw* | \
-    i?86*-*-cygwin* | x86_64*-*-cygwin*)
+    aarch64-*-mingw* | i?86*-*-mingw* | x86_64*-*-mingw* | \
+    aarch64-*-cygwin* | i?86*-*-cygwin* | x86_64*-*-cygwin*)
       enable_decimal_float=yes
       ;;
     *)
diff --git a/configure.ac b/configure.ac
index 05ddf6987..7e8a6b1c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -869,7 +869,7 @@ case "${target}" in
   i[[3456789]]86-*-mingw*)
     noconfigdirs="$noconfigdirs ${libgcj}"
     ;;
-  x86_64-*-mingw*)
+  aarch64-*-mingw* | x86_64-*-mingw*)
     noconfigdirs="$noconfigdirs ${libgcj}"
     ;;
   mmix-*-*)
@@ -3225,7 +3225,7 @@ case " $target_configdirs " in
 esac
 
 case "$target" in
-  x86_64-*mingw* | *-w64-mingw*)
+  aarch64-*mingw* | x86_64-*mingw* | *-w64-mingw*)
   # MinGW-w64 does not use newlib, nor does it use winsup. It may,
   # however, use a symlink named 'mingw' in ${prefix} .
     FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/mingw/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/mingw/include'
